libraries/AP_GPS/AP_GPS_UBLOX.cpp

```diff
@@ -234,11 +233,21 @@ AP_GPS_UBLOX::_verify_rate(uint8_t msg_class, uint8_t msg_id, uint8_t rate) {
                 _cfg_needs_save = true;
             }
             break;
+        case MSG_PVT:
+            if(rate == RATE_PVT) {
+                _unconfigured_messages &= ~CONFIG_RATE_PVT;
+            } else {
+                _configure_message_rate(msg_class, msg_id, RATE_PVT);
+                _unconfigured_messages |= CONFIG_RATE_PVT;
+                _cfg_needs_save = true;
+            }
```

Comment:
Yeah, at the same time I think almost all the other drivers are entirely dependent on the initial config string for getting the correct state, or their constructor. As far as I know u-blox is the only GPS driver we have that can recover every setting except for baud rate while the GPS is connected, so for most drivers the current behaviour is the most assistance they can get, while in the case of u-blox it isn't helping us.